ql/math/integrals/gausslobattointegral.cpp

```diff
@@ -138,7 +138,7 @@ namespace QuantLib {
         
         // avoid 80 bit logic on x86 cpu
         volatile Real dist = acc + (integral1-integral2);
-        if(Real(dist)==acc || mll<=a || b<=mrr) {
+        if(const_cast<Real&>(dist)==acc || mll<=a || b<=mrr) {
```

Comment:
I'm not sure I understand this one?

Reply:
We don't fully understand why `volatile` was needed in the original code. From a math perspective, `dist == acc` if `integral1 == integral2` (however, exact equality is quite unlikely given how those terms are computed above, and equality comparison of floating point numbers is usually unstable). But this code has been in QuantLib for 14 years (the line with `volatile`) and we can't fully assess the implications of removing/rewriting this part of the code. 

Regarding the change: The cast to `Real` in the original code casts the `volatile` away using a C-style cast - we simply replaced it with the correct C++ cast for removing the `volatile` qualifier: `const_cast`.